### PR TITLE
Send quote-enclosed pipeline name to API when retrieving pipeline.

### PR DESCRIPTION
**Pull request by robsyme**
Signed-off-by: Robert Syme <rob.syme@gmail.com>

### CODE REVIEW

**Comment by robsyme** [comment]
Closes https://github.com/seqeralabs/tower-cli/issues/360

File: src/main/java/io/seqera/tower/cli/commands/pipelines/AbstractPipelinesCmd.java [hunk 1]
```diff
@@ -37,7 +37,7 @@ public AbstractPipelinesCmd() {
 
     protected PipelineDbDto pipelineByName(Long workspaceId, String name) throws ApiException {
 
-        ListPipelinesResponse list = api().listPipelines(Collections.emptyList(), workspaceId, null, null, name, "all");
+        ListPipelinesResponse list = api().listPipelines(Collections.emptyList(), workspaceId, null, null, "\"" + name + "\"", "all");
```
Comment (jordeu) [inline src/main/java/io/seqera/tower/cli/commands/pipelines/AbstractPipelinesCmd.java:40]: Better check that is not already quoted and escape any other quote.

Here a ChatGPT proposal:
```java
public class StringQuoter {
    public static String quoteString(String input) {
        if (input == null) {
            return "\"\"";
        }

        // Escaping intermediate quotes
        String escapedInput = input.replace("\"", "\\\"");

        // Check if the string is already quoted
        if (escapedInput.startsWith("\"") && escapedInput.endsWith("\"")) {
            return escapedInput;
        } else {
            // Adding quotes
            return "\"" + escapedInput + "\"";
        }
    }

    public static void main(String[] args) {
        // Test the function
        String testString = "Hello, \"world\"!";
        System.out.println(quoteString(testString)); // Should print: "Hello, \"world\"!"

        String alreadyQuoted = "\"Hello, world!\"";
        System.out.println(quoteString(alreadyQuoted)); // Should print the same string without changes
    }
}
```

**Comment by pditommaso** [comment]
Not sure it's a good idea, why doing this?

**Comment by jordeu** [comment]
> Not sure it's a good idea, why doing this?

What do you mean? my comment or the whole thing?

**Comment by jordeu** [comment]
The root of the problem is that here the CLI wants to do an exact match search by pipeline name because we expect the name to be unique. And the only way of doing this at API level is enclosing the name with quotes.

**Comment by jordeu** [comment]
> I see; therefore, it's the field in the request. As such, should not be managed on BE side? how the FE handle this?

The FE do not has this need because it is always first doing a list of all the pipelines in a workspace and from that list knows the internal Id of the pipeline. So FE is never querying BE for a single pipeline by name.

**Comment by JaimeSeqLabs** [comment]
Closing this in favor of https://github.com/seqeralabs/tower-cli/pull/425

**Pull request closed without merging**